### PR TITLE
Fix

### PR DESCRIPTION
**Pull request by MACE4GITHUB**
Changed Address (inc. Location). Renamed DTO. Removed absolute paths. Renamed folder IDP to IdentityServer. Fix reference packages for docker.

### CODE REVIEW

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/DiscountProgramAdministratorResponse.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ namespace Exadel.CrazyPrice.Common
     /// <summary>
     /// Gets full information for an Administrator.
     /// </summary>
-    public class DiscountProgramWithDateModify
+    public class DiscountProgramAdministratorResponse
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/DiscountProgramAdministratorResponse.cs:9]: Please check that this separate classes will be needed. Probably we can use same class with empty fields and setup rules to serialize to JSON with empty fields

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/NewDiscountProgramModerator.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ namespace Exadel.CrazyPrice.Common
     /// <summary>
     /// Gets/sets information for a Moderator
     /// </summary>
-    public class NewDiscountProgram
+    public class NewDiscountProgramModerator
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/NewDiscountProgramModerator.cs:9]: Probably rename to NewDiscountProgramRequest

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/DiscountProgramAdministratorResponse.cs (hunk 1 above)
Comment (MACE4GITHUB) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/DiscountProgramAdministratorResponse.cs]: It fixed [https://github.com/yegorl/exadel_discounts_be/pull/3/commits/842b9c9fd0ced13302bc339c977307d14e491896](url)

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/NewDiscountProgramModerator.cs (hunk 2 above)
Comment (MACE4GITHUB) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/NewDiscountProgramModerator.cs]: [It fixed](https://github.com/yegorl/exadel_discounts_be/blob/fix/src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/NewDiscountProgramRequest.cs)

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Extentions/StringExtention.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Exadel.CrazyPrice.Common.Extentions
+{
+    /// <summary>
+    /// Represents correcting string.
+    /// </summary>
+    public static class StringExtention
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Extentions/StringExtention.cs:8]: Rename to StringExtenstions - probably one more to appear

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+namespace Exadel.CrazyPrice.Common.Interfaces
+{
+    public interface ICrazyPriceRepository
+    {        
+
+        /// <summary>
+        /// Gets all Discount Programs.
+        /// </summary>
+        /// <returns></returns>
+        Task<IEnumerable<DiscountResponse>> GetAllDiscountProgramsAsync();
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs:15]: Probably List<> will be better? IEnumerable may cause delayed LINQ queries with unpredictive results

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        /// <param name="id">The id Discount Program.</param>
+        /// <returns></returns>
+        Task<DiscountResponse> GetDiscountProgramAsync(string id);
+
+        /// <summary>
+        /// Gets a Discount for Moderator Program by id Discount Program.
+        /// </summary>
+        /// <param name="id">The id Discount Program.</param>
+        /// <returns></returns>
+        Task<UpsertDiscountRequest> GetDiscountProgramForModeratorAsync(string id);
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs:29]: Method names strange - does it Get from database or request for upsert

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+        /// <summary>
+        /// Creates or updates a Discount Program by Moderator.
+        /// </summary>
+        /// <param name="item"></param>
+        /// <returns></returns>
+        Task UpdateDiscountProgramAsync(UpsertDiscountRequest item);
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs:36]: Make Upsert to have similar naming

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    {      
+        /// <summary>
+        /// Gets or sets a search text.
+        /// </summary>
+        public string SearchText { get; set; }
+
+        /// <summary>
+        /// Gets or sets a Country.
+        /// </summary>
+        public string Country { get; set; }
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs:13]: Make a SearhcCountry, SearchCity

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+        /// <summary>
+        /// Gets or sets a City.
+        /// </summary>
+        public string City { get; set; }
+
+        /// <summary>
+        /// Gets or sets a Discount Program option.
+        /// </summary>
+        public DiscountOption DiscountOption { get; set; }
```
Comment (yegorl) [inline src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs:23]: SearchDiscountOption

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Extentions/StringExtention.cs (hunk 3 above)
Comment (MACE4GITHUB) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Extentions/StringExtention.cs]: [Fix it](https://github.com/yegorl/exadel_discounts_be/blob/fix/src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Extentions/StringExtentions.cs)

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs (hunk 4 above)
Comment (MACE4GITHUB) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs]: [Fix it](https://github.com/yegorl/exadel_discounts_be/blob/fix/src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/IDiscountRepository.cs)

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs (hunk 5 above)
Comment (MACE4GITHUB) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs]: [Fix it](https://github.com/yegorl/exadel_discounts_be/blob/fix/src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/IDiscountRepository.cs)

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs (hunk 6 above)
Comment (MACE4GITHUB) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/ICrazyPriceRepository.cs]: [Fix it](https://github.com/yegorl/exadel_discounts_be/blob/fix/src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Interfaces/IDiscountRepository.cs)

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs (hunk 7 above)
Comment (MACE4GITHUB) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs]: [Fix it](https://github.com/yegorl/exadel_discounts_be/blob/fix/src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs)

File: src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs (hunk 8 above)
Comment (MACE4GITHUB) [reply src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs]: [Fix it](https://github.com/yegorl/exadel_discounts_be/blob/fix/src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/Models/SearchCriteria.cs)

**Review by yegorl — APPROVED** [review]

**Pull request merged**